File: 김찬우/프로그래머스 /Lv.2/k진수에서 소수 개수 구하기.java
import java.lang.Math;
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        
        String a = Integer.toString(n, k);
    
        String arr[] = a.split("0");
        
        for(String str: arr){
            
            if(str.equals("")){
                continue;
            }
            
            if(check(Long.parseLong(str))){
                answer++;
            }
            
        }

        
        
        return answer;
    }
    
    public boolean check(long num){
        
        if(num == 1){
            return false;
        }
        
        for(int i=2; i<=(int)Math.sqrt(num); i++){
            
            if(num % i == 0){
                
                return false;
                
            }
            
        }
        
        return true;
        
    }
        
}

File: 조준장/level2/k진수에서 소수 개수 구하기.kt
import java.math.*

class Solution {
    fun solution(n: Int, k: Int): Int {
        var answer: Int = 0
        n.toString(k)
            .split("0")
            .forEach {
            if (it == "" || it == "0" || it == "1") return@forEach
            if (BigInteger(it).isProbablePrime(1)) answer++
        }
        return answer
    }
}
